src/common/client-app.ts

```diff
@@ -21,20 +21,28 @@ export var defaultConfiguration: AppConfiguration = {
 
 export function start(boardVM: IBoardVM, config?: AppConfiguration) {
   var app = new App(boardVM, config);
-  app.start();
+  config = utils.extend({ }, defaultConfiguration, config);
+  boardVM.applyBindings(config.rootNode);
+  var socket = io();
+  socket.on(config.socketEventName, app.onMessage);
+
+  app.sendToServer = (changes) => {
+      socket.emit(config.socketEventName, { patch: changes });
+  };
+  setInterval(app.onInterval, config.throttlingInterval);
   return app;
 }
 
-class App {
+export class App {
   shadowServer: model.Board = { };
   shadowClient: model.Board = { };
   boardVM: IBoardVM;
-  socket: SocketIOClient.Socket;
   config: AppConfiguration;
 
   constructor(boardVM: IBoardVM, config?: AppConfiguration) {
     this.boardVM = boardVM;
     this.config = utils.extend({ }, defaultConfiguration, config);

```

Comment:
Oh, the configuration is not needed anymore in  App class, what do you think about remove the parameter?